subsys/usb/class/bluetooth.c

```diff
@@ -126,6 +126,20 @@ static void hci_tx_thread(void)
 		struct net_buf *buf;
 
 		buf = net_buf_get(&tx_queue, K_FOREVER);
+		if (suspended) {
+			if (usb_wakeup_request()) {
+				LOG_DBG("Remote wakeup not enabled/supported");
+			}
+			/*
+			 * Let's wait until operation is resumed.
+			 * This is independent of usb_wakeup_request() result,
+			 * as long as device is suspended it should not start
+			 * any transfers.
+			 */
+			while (suspended) {
+				k_sleep(K_MSEC(1));
+			}
```

Comment:
1. I assume `suspended` is volatile?
2. Can `k_sem_take` construct be used instread of `k_sleep` wait?

Reply:
> Can k_sem_take construct be used instread of k_sleep wait?

Why and for what benefit?

Reply:
To avoid having to wake up periodically to check. Instead you'd have some other code asynchronously post `k_sem_give()`.

Reply:
The `volatile` is necessary if `suspended` is being set from a different context that can be preempted or that can preempt, so I agree with @cvinayak that this might be required.

Reply:
As we are in a while loop, without `suspended` being quailified as volatile would mean a register storage will be used and no fresh memory access will happen, potentially causing a deadlock in this while loop.

Use of semaphore signal  on change to `suspended` will avoid an upto 1 ms latency in the USB transport on wakeup.

Reply:
>To avoid having to wake up periodically to check. Instead you'd have some other code asynchronously post k_sem_give().

This some code would add unnecessary complexity.

Reply:
It is not complex and most simplest synchronization (there many more such use across zephyr, I believe), see for example here: https://github.com/zephyrproject-rtos/zephyr/blob/a3c91034cd5aa54c5f605044ee7bbacfd395c0be/drivers/clock_control/clock_control_nrf.c#L381-L406